includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -2552,6 +2559,11 @@ public function doing_ajax() {
 			return true;
 		}
 
+		// To capture the cart redirect on update cart actions.
+		if ( isset( $_SERVER['HTTP_REFERER'] ) && $_SERVER['HTTP_REFERER'] === wc_get_cart_url() ) {
```

Comment:
Did we consider using `wp_get_referer` function here?

Reply:
Yes, unfortunately it didn't work. Adding 
```
var_dump( wp_get_referer(), $_SERVER['HTTP_REFERER'], wp_get_referer() === $_SERVER['HTTP_REFERER'] );
die();
```

above the line 2562 produces:

bool(false)
string(34) "{proto}://{domain}/cart/"
bool(false)

Reply:
I see. Can we add a comment to mention we considered it, but it didn't work?

Reply:
Yes, added a comment also fixed a phpcs about Yoda contitionals that was affecting that line.

Reply:
`wp_get_referer()` returns false if referred by the same page, as per documentation:

> If page “refered” (form posted) to itself, returns false (because $_SERVER[‘HTTP_REFERER’] == $_REQUEST[‘_wp_http_referer’])

`wp_get_raw_referer()` is available and returns header contents without validation.

Reply:
So, actually you might want to replace if condition with `if ( false === wp_get_referer() )`.

Reply:
Hey @dmvrtx, yea that seems like a good idea. Gave it a try and everything seems to be working as expected. The desired functionality in the cart page is there after clicking the Update Cart button and i couldn't see any side effects. Created a draft [PR](https://github.com/woocommerce/woocommerce-gateway-amazon-pay/pull/218) in order to include it in next release.